connectors/file/translator-parquet/src/main/java/org/teiid/translator/parquet/ParquetQueryVisitor.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags and
+ * the COPYRIGHT.txt file distributed with this work.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.teiid.translator.parquet;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Stack;
+
+import org.teiid.language.*;
+import org.teiid.language.visitor.HierarchyVisitor;
+import org.teiid.metadata.Column;
+import org.teiid.metadata.Table;
+import org.teiid.translator.TranslatorException;
+
+public class ParquetQueryVisitor extends HierarchyVisitor {
+
+    protected Stack<Object> onGoingExpression = new Stack<Object>();
+    private List<String> projectedColumnNames = new ArrayList<String>();
+    protected ArrayList<TranslatorException> exceptions = new ArrayList<TranslatorException>();
+
+    public HashMap<String, Comparison> getColumnPredicates() {
```

Comment:
Now that ordered predicates are supported the strategy of mapping to a single Comparison won't work.  For example add a test that has a predicate like WHERE id > 2 and id < 4.  You don't need to check for the case of an equality predicate and something else - the engine will always reduce down to a single equality predicate when possible.

Reply:
> Now that ordered predicates are supported the strategy of mapping to a single Comparison won't work

Yeah, what do we need to do then, a multimap?

>You don't need to check for the case of an equality predicate and something else - the engine will always reduce down to a single equality predicate when possible.

Sorry. but I didn't get this. where am I checking for an equality predicate?

Reply:
> Yeah, what do we need to do then, a multimap?

Yes a multi-map will work.

> Sorry. but I didn't get this. where am I checking for an equality predicate?

I just mean you don't have to do any additional special checks at the translator level or try to logically combine predicates.